### PR TITLE
Tree Upload Reminder Dialog

### PR DESCRIPTION
**Pull request by Mzazi25**
This PR adds Alert Dialog to users who have planted trees above 2000 and reminds them to upload the trees. 
Fixes on Issue #1057

### CODE REVIEW

File: app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardScreen.kt [hunk 1]
```diff
@@ -63,13 +61,30 @@ fun DashboardScreen(
     val state by viewModel.state.observeAsState(DashboardState())
     val snackBar by viewModel.snackBar.observeAsState()
     val navController = LocalNavHostController.current
+    var dialogOpened by remember { mutableStateOf(false) }
+    var showDialog by remember { mutableStateOf(false) }
 
+    if (showDialog){
+        DisplayAlertDialog(
+            title = "Upload Trees Soon",
+            message ="You have over 2000 trees, please upload as soon as you can.",
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardScreen.kt:70]: Text that the user sees should always use string resources to allow translations.

File: app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardScreen.kt [hunk 2]
```diff
@@ -63,13 +61,30 @@ fun DashboardScreen(
...
+        )
+    }
     Dashboard(
         state = state,
         snackBar = snackBar,
         onSyncClicked = { viewModel.sync() },
         onOrgClicked = { navController.navigate(NavRoute.Org.route) },
-        onCaptureClicked = { navController.navigate(NavRoute.UserSelect.route) },
+        onCaptureClicked = {
+            if (state.totalTreesToSync >=2000){
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardScreen.kt:84]: We can move this check into a boolean method on the state object that returns true or false

File: app/src/main/java/org/greenstand/android/TreeTracker/dashboard/components/DisplayAlertDialog.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.Button
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.sp
+
+@Composable
+fun DisplayAlertDialog(
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/dashboard/components/DisplayAlertDialog.kt:11]: We have a dialog method we can reuse that has a special style that matches the app. Please use that instead

**Review by Elforama — CHANGES_REQUESTED** [review]
Thanks for tackling this. There are some required changes, but it looks good overall.

Please keep formatting in mind while working, try to match the style in the file.

File: app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardScreen.kt [hunk 4]
```diff
@@ -63,13 +61,29 @@ fun DashboardScreen(
     val state by viewModel.state.observeAsState(DashboardState())
     val snackBar by viewModel.snackBar.observeAsState()
     val navController = LocalNavHostController.current
+    var dialogOpened by remember { mutableStateOf(false) }
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardScreen.kt:64]: dialogOpened doesn't seem to be used

File: app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardViewModel.kt [hunk 5]
```diff
@@ -36,6 +36,7 @@ data class DashboardState(
     val totalTreesToSync: Int = 0,
     val isOrgButtonEnabled: Boolean = false,
     val showUnreadMessageNotification: Boolean = false,
+    val totalTreesToSyncThreshold:Boolean = false
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardViewModel.kt:39]: This should be renamed to something more descriptive like "showTreeSyncReminderDialog"

**Review by Elforama — CHANGES_REQUESTED** [review]
Thanks for making the changes, it's looking a lot better. Can you please cleanup the last few things before I merge it?

File: app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardScreen.kt (hunk 4 above)
Comment (Mzazi25) [reply app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardScreen.kt]: fixed

File: app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardViewModel.kt (hunk 5 above)
Comment (Mzazi25) [reply app/src/main/java/org/greenstand/android/TreeTracker/dashboard/DashboardViewModel.kt]: fixed

**Review by Elforama — APPROVED** [review]

**Pull request merged**